### PR TITLE
e2e: Refactor the CatalogSource tests to reduce flakes

### PR DESCRIPTION
**Pull request by timflannagan**
**Description of the change:**
Follow up to #2690 as it doesn't look like I have the right permissions to push to Per's fork.

**Motivation for the change:**
See #2690 for more information.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky
- [ ] Tests that remove the `[FLAKE]` tag are no longer flaky

### CODE REVIEW

File: test/e2e/catalog_e2e_test.go [hunk 1]
```diff
@@ -138,15 +147,16 @@ var _ = Describe("Catalog represents a store of bundles which OLM can use to ins
 		}
 
 		// Create the initial catalog source
-		createInternalCatalogSource(c, crc, mainCatalogName, globalNS, mainManifests, []apiextensions.CustomResourceDefinition{mainCRD}, []v1alpha1.ClusterServiceVersion{mainCSV})
+		cs, cleanup := createInternalCatalogSource(c, crc, mainCatalogName, globalNS, mainManifests, []apiextensions.CustomResourceDefinition{mainCRD}, []v1alpha1.ClusterServiceVersion{mainCSV})
+		defer cleanup()
```
Comment (timflannagan) [inline test/e2e/catalog_e2e_test.go:151]: Note: only substantial difference between Per's original PR and this one. I noticed we weren't cleaning up this CatalogSource that's deployed in the global namespace correctly.

File: test/e2e/catalog_e2e_test.go [hunk 2]
```diff
@@ -64,14 +74,14 @@ var _ = Describe("Catalog represents a store of bundles which OLM can use to ins
 		}
 
 		crd := newCRD(genName("ins"))
-		csv := newCSV(packageStable, testNamespace, "", semver.MustParse("0.1.0"), []apiextensions.CustomResourceDefinition{crd}, nil, nil)
+		csv := newCSV(packageStable, ns.GetName(), "", semver.MustParse("0.1.0"), []apiextensions.CustomResourceDefinition{crd}, nil, nil)
 
 		catalogSourceName := genName("mock-ocs-")
-		_, cleanupSource := createInternalCatalogSource(c, crc, catalogSourceName, operatorNamespace, manifests, []apiextensions.CustomResourceDefinition{crd}, []v1alpha1.ClusterServiceVersion{csv})
+		_, cleanupSource := createInternalCatalogSource(c, crc, catalogSourceName, ns.GetName(), manifests, []apiextensions.CustomResourceDefinition{crd}, []v1alpha1.ClusterServiceVersion{csv})
```
Comment (timflannagan) [inline test/e2e/catalog_e2e_test.go:80]: Note: I noticed this test case was missed when ensuring that CatalogSources get deployed in their own namespace (outside of any tests that are testing the global namespace behavior), so any change you see for the remainder of this test spec's diff is substituting operatorNamespace -> the generated namespace's metadata.Name.

**Comment by timflannagan** [comment]
Hmm interesting:

```
• Failure [323.808 seconds]
Catalog represents a store of bundles which OLM can use to install Operators
/home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/catalog_e2e_test.go:38
  delete internal registry pod triggers recreation [It]
  /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/catalog_e2e_test.go:539

  
  	Error Trace:	util.go:110
  	            				catalog_e2e_test.go:596
  	            				runner.go:113
  	            				runner.go:64
  	            				it_node.go:26
  	            				spec.go:215
  	            				spec.go:138
  	            				spec_runner.go:200
  	            				spec_runner.go:170
  	            				spec_runner.go:66
  	            				suite.go:79
  	            				ginkgo_dsl.go:245
  	            				ginkgo_dsl.go:228
  	            				e2e_test.go:59
  	Error:      	Received unexpected error:
  	            	timed out waiting for the condition
  	Test:       	Catalog represents a store of bundles which OLM can use to install Operators delete internal registry pod triggers recreation
```

https://github.com/operator-framework/operator-lifecycle-manager/runs/5545060704?check_suite_focus=true

**Comment by akihikokuroda** [comment]
^^^ may be related to 2 registry pods and probably fixed by https://github.com/operator-framework/operator-lifecycle-manager/pull/2614#discussion_r808014164

**Comment by timflannagan** [comment]
https://github.com/operator-framework/operator-lifecycle-manager/pull/2693#issuecomment-1067360442 x2

re-spinning again.

**Pull request merged**